### PR TITLE
Map World.class

### PR DESCRIPTION
**Pull request by matthewperiut**
just a start for usability, will probably need lots of revision

### CODE REVIEW

File: mappings/net/minecraft/entity/player/PlayerEntity.mapping [hunk 1]
```diff
@@ -28,8 +29,10 @@ CLASS net/minecraft/class_54 net/minecraft/entity/player/PlayerEntity
 		ARG 2 amount
 	METHOD method_492 respawn ()V
 	METHOD method_493 closeScreen ()V
+	METHOD method_497 stopSleeping (ZZZ)V
```
Comment (DanyGames2014) [inline mappings/net/minecraft/entity/player/PlayerEntity.mapping]: [wakeUp](https://maven.fabricmc.net/docs/yarn-1.20-pre4+build.2/net/minecraft/entity/player/PlayerEntity.html#wakeUp(boolean,boolean)) in Yarn

**Comment by matthewperiut** [comment]
this is ready for review, maps all functions and fields of World except for two which are in #31 instead.

**Review by DanyGames2014 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/player/PlayerEntity.mapping (hunk 1 above)
Comment (matthewperiut) [reply mappings/net/minecraft/entity/player/PlayerEntity.mapping]: fixed

File: mappings/net/minecraft/world/World.mapping [hunk 2]
```diff
@@ -1,16 +1,57 @@
...
+	FIELD field_188 allPlayersSleeping Z
+	FIELD field_189 tempCollisionBoxes Ljava/util/ArrayList;
+	FIELD field_190 processingDeferred Z
+	FIELD field_191 lightingUpdateCount I
+	FIELD field_192 allowMonsterSpawning Z
+	FIELD field_193 allowMobSpawning Z
+	FIELD field_194 activeChunks Ljava/util/Set;
+	FIELD field_195 ambientSoundCounter I
+	FIELD field_196 tempEntityList Ljava/util/List;
+	FIELD field_197 instantTick Z
```
Comment (telvarost) [inline mappings/net/minecraft/world/World.mapping]: The MCP name for this is `scheduledUpdatesAreImmediate` should we change the name of this to reflect that it is a boolean rather than an integer tick count. Would it be better named `instantTickingEnabled` or something?

File: mappings/net/minecraft/world/World.mapping [hunk 3]
```diff
@@ -36,41 +159,156 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
...
 	METHOD method_191 playSound (Lnet/minecraft/class_57;Ljava/lang/String;FF)V
-	METHOD method_200 (IIII)Z
+		ARG 1 source
+		ARG 2 sound
+	METHOD method_192 (Lnet/minecraft/class_57;Lnet/minecraft/class_57;F)Lnet/minecraft/class_61;
+		ARG 1 entity
+	METHOD method_193 updateEntity (Lnet/minecraft/class_57;Z)V
+		ARG 1 entity
+	METHOD method_194 processScheduledTicks (Z)Z
+	METHOD method_195 saveDisplay (ZLnet/minecraft/class_62;)V
```
Comment (telvarost) [inline mappings/net/minecraft/world/World.mapping]: Should this be renamed to `saveWithLoadingDisplay` to better indicate that this function actually saves the world and doesn't just bring up the save screen?

File: mappings/net/minecraft/client/gui/screen/LoadingDisplay.mapping [hunk 4]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/class_62 net/minecraft/client/gui/screen/LoadingDisplay
```
Comment (telvarost) [inline mappings/net/minecraft/client/gui/screen/LoadingDisplay.mapping:1]: Hmm I guess this is what it is in modern. In beta it is really just for a loading progress bar, I believe. I guess it will be more clear when the other fields eventually get mapped.

File: mappings/net/minecraft/world/World.mapping [hunk 5]
```diff
@@ -80,30 +318,192 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
...
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
+	METHOD method_247 camTransferPowerInDirection (IIII)Z
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
+		ARG 4 direction
+	METHOD method_248 manageChunkUpdatesAndEvents ()V
+	METHOD method_249 canPlaceAt (III)Z
```
Comment (telvarost) [inline mappings/net/minecraft/world/World.mapping]: MCP has this name as `canBlockSeeTheSky` probably just a misnomer on MCP's part as I don't immediately notice anything about this function that references seeing the sky. But the code in this function is very hard to parse what is happening. I guess the function is trying to decide if the block you are placing is above world height or not? Is that your understanding of what is going on here?

File: mappings/net/minecraft/world/World.mapping [hunk 6]
```diff
@@ -80,30 +318,192 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
...
+		ARG 3 z
+	METHOD method_253 neighborUpdate (IIII)V
+		ARG 1 x
+		ARG 2 y
 		ARG 3 z
+		ARG 4 blockId
 	METHOD method_254 getSeed ()J
+	METHOD method_255 getLightLevel (III)I
 	METHOD method_256 getTime ()J
+	METHOD method_257 isTopY (III)Z
```
Comment (telvarost) [inline mappings/net/minecraft/world/World.mapping:446]: It looks like this is the actual function for determining if a block is exposed to the sky, by finding the highest Y block, or something. Idk. In MCP its called `canExistingBlockSeeTheSky`. There is a function in modern yarn called `isTopSolid` I guess maybe this is similar?

File: mappings/net/minecraft/world/World.mapping [hunk 7]
```diff
@@ -80,30 +318,192 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
 		ARG 6 sizeZ
 	METHOD method_218 isBoxSubmergedInFluid (Lnet/minecraft/class_25;)Z
 		ARG 1 box
+	METHOD method_219 loadEntitySkin (Lnet/minecraft/class_57;)V
+		ARG 1 entity
+	METHOD method_220 canMonsterSpawn ()Z
+	METHOD method_221 getRainGradient (F)F
+		ARG 1 delta
 	METHOD method_222 getTopY (II)I
```
Comment (telvarost) [inline mappings/net/minecraft/world/World.mapping:338]: At this point maybe its best to just ignore my top Y block comments. In MCP its `getHeightValue` I guess i haven't really used any of these functions at all. So I don't get the difference between them all that much

File: mappings/net/minecraft/world/event/listener/GameEventListener.mapping [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+		ARG 10 velocityY
+		ARG 12 velocityZ
+	METHOD method_1154 playSound (Ljava/lang/String;DDDFF)V
+		ARG 1 sound
+		ARG 2 x
+		ARG 4 y
+		ARG 6 z
+		ARG 8 volume
+		ARG 9 pitch
+	METHOD method_1155 playMusic (Ljava/lang/String;III)V
```
Comment (telvarost) [inline mappings/net/minecraft/world/event/listener/GameEventListener.mapping]: This seems incorrect. This would make me think that it would play background music from `/music` or `/newmusic` directories. When it actually plays from `/streaming` because this is the function for playing records I believe.

File: mappings/net/minecraft/world/World.mapping [hunk 9]
```diff
@@ -36,41 +159,156 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
...
+		ARG 1 entityClass
+		ARG 2 box
 	METHOD method_176 getOrCreateState (Ljava/lang/Class;Ljava/lang/String;)Lnet/minecraft/class_58;
 		ARG 1 stateClass
 		ARG 2 id
+	METHOD method_177 getPlayer (Ljava/lang/String;)Lnet/minecraft/class_54;
+		ARG 1 name
 	METHOD method_178 addParticle (Ljava/lang/String;DDDDDD)V
+		ARG 1 particle
+	METHOD method_179 playMusic (Ljava/lang/String;III)V
```
Comment (telvarost) [inline mappings/net/minecraft/world/World.mapping]: This seems incorrect. This would make me think that it would play background music from `/music` or `/newmusic` directories. When it actually plays from `/streaming` because this is the function for playing records I believe. The names of both functions should be `playStreaming` or `playRecord` or something

File: mappings/net/minecraft/world/World.mapping [hunk 10]
```diff
@@ -80,30 +318,192 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
...
+		ARG 2 y
+		ARG 3 z
+		ARG 4 soundType
+		ARG 5 pitch
+	METHOD method_225 isFireOrLavaInBox (Lnet/minecraft/class_25;)Z
+		ARG 1 box
+	METHOD method_226 unloadEntitySkin (Lnet/minecraft/class_57;)V
+		ARG 1 entity
+	METHOD method_227 tickEntities ()V
+	METHOD method_228 getTopSolidBlockY (II)I
```
Comment (telvarost) [inline mappings/net/minecraft/world/World.mapping:357]: The legendary function here it is at last! This is the one that people would actually use I think. It's name seems good 😃

File: mappings/net/minecraft/world/World.mapping [hunk 11]
```diff
@@ -1,16 +1,57 @@
...
+	FIELD field_212 saveInterval I
+	FIELD field_213 difficulty I
+	FIELD field_214 random Ljava/util/Random;
+	FIELD field_215 newWorld Z
 	FIELD field_216 dimension Lnet/minecraft/class_50;
+	FIELD field_217 eventListeners Ljava/util/List;
+	FIELD field_218 chunkManager Lnet/minecraft/class_51;
 	FIELD field_219 dimensionData Lnet/minecraft/class_52;
 	FIELD field_220 properties Lnet/minecraft/class_7;
+	FIELD field_221 isEventProcessingEnabled Z
```
Comment (telvarost) [inline mappings/net/minecraft/world/World.mapping]: This is one of the few places with the `isAction` naming convention is used do we want to switch this not to have `isAction` to match modern yarn boolean naming convention for fields (getters use the `isAction` naming convention instead in modern yarn), so the new name should be `eventProcessingEnabled`

File: mappings/net/minecraft/world/World.mapping [hunk 12]
```diff
@@ -20,12 +61,94 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
...
+		ARG 1 type
+		ARG 2 minX
+		ARG 3 minY
+		ARG 4 minZ
+		ARG 5 maxX
+		ARG 6 maxY
+		ARG 7 maxZ
 	METHOD method_168 canSpawnEntity (Lnet/minecraft/class_25;)Z
 		ARG 1 box
 	METHOD method_169 isBoxInMaterial (Lnet/minecraft/class_25;Lnet/minecraft/class_15;)Z
```
Comment (telvarost) [inline mappings/net/minecraft/world/World.mapping]: Should this be flip flopped to `isMaterialInBox` this is the way it is for MCP `isMaterialInBB`

File: mappings/net/minecraft/world/World.mapping [hunk 13]
```diff
@@ -80,30 +318,192 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
...
+	METHOD method_263 canTransferPower (III)Z
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
+	METHOD method_264 updateSleepingPlayers ()V
+	METHOD method_265 isEmittingRedstonePower (III)Z
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
+	METHOD method_266 skipNight ()V
```
Comment (telvarost) [inline mappings/net/minecraft/world/World.mapping]: This function just wakes up all the players and clears weather, it doesn't really change world time at all. Could this be renamed to `onNightFinished` or `onSleepFinished` or something. In MCP the name is `wakeUpAllPlayers`

Comment (telvarost) [reply mappings/net/minecraft/world/World.mapping]: I think we decided on call that `afterSkipNight` would be a good name for it

File: mappings/net/minecraft/world/explosion/Explosion.mapping [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+	FIELD field_1392 x D
+	FIELD field_1393 y D
+	FIELD field_1394 z D
+	FIELD field_1395 source Lnet/minecraft/class_57;
+	FIELD field_1396 power F
+	FIELD field_1398 random Ljava/util/Random;
+	FIELD field_1399 world Lnet/minecraft/class_18;
+	METHOD <init> (Lnet/minecraft/class_18;Lnet/minecraft/class_57;DDDF)V
+		ARG 1 world
+		ARG 2 soruce
```
Comment (matthewperiut) [inline mappings/net/minecraft/world/explosion/Explosion.mapping]: should be "source" misspelling

File: mappings/net/minecraft/world/World.mapping (hunk 2 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/World.mapping]: `instantBlockUpdateEnabled` probably

File: mappings/net/minecraft/world/World.mapping (hunk 3 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/World.mapping]: will do

File: mappings/net/minecraft/world/World.mapping (hunk 6 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/World.mapping:434]: will be isTopSolid

Comment (matthewperiut) [reply mappings/net/minecraft/world/World.mapping:446]: probably fine as is

File: mappings/net/minecraft/world/event/listener/GameEventListener.mapping (hunk 8 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/event/listener/GameEventListener.mapping]: I'll make it playStreaming

File: mappings/net/minecraft/world/World.mapping (hunk 9 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/World.mapping]: I'll make it playStreaming

File: mappings/net/minecraft/world/World.mapping (hunk 11 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/World.mapping]: will be `eventProcessingEnabled`

File: mappings/net/minecraft/world/World.mapping (hunk 12 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/World.mapping]: will be `isMaterialInBox`

File: mappings/net/minecraft/world/World.mapping (hunk 13 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/World.mapping]: `afterSkipNight` sounds good

File: mappings/net/minecraft/world/World.mapping (hunk 5 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/World.mapping]: will be `hasSkyLight`

**Review by telvarost — APPROVED** [review]

File: mappings/net/minecraft/world/chunk/light/LightStorage.mapping [hunk 15]
```diff
@@ -0,0 +1,18 @@
+CLASS net/minecraft/class_417 net/minecraft/world/chunk/light/LightStorage
+	FIELD field_1673 type Lnet/minecraft/class_56;
```
Comment (DanyGames2014) [inline mappings/net/minecraft/world/chunk/light/LightStorage.mapping]: [lightType](https://maven.fabricmc.net/docs/yarn-1.20.6+build.3/net/minecraft/world/chunk/light/LightStorage.html#lightType) in Yarn

File: mappings/net/minecraft/world/chunk/light/LightStorage.mapping [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+	FIELD field_1674 minX I
+	FIELD field_1675 minY I
+	FIELD field_1676 minZ I
+	FIELD field_1677 maxX I
+	FIELD field_1678 maxY I
+	FIELD field_1679 maxZ I
+	METHOD <init> (Lnet/minecraft/class_56;IIIIII)V
+		ARG 1 type
```
Comment (DanyGames2014) [inline mappings/net/minecraft/world/chunk/light/LightStorage.mapping]: lightType to be consistent with field_1673

File: mappings/net/minecraft/world/chunk/light/LightStorage.mapping [hunk 17]
```diff
@@ -0,0 +1,18 @@
+CLASS net/minecraft/class_417 net/minecraft/world/chunk/light/LightStorage
```
Comment (DanyGames2014) [inline mappings/net/minecraft/world/chunk/light/LightStorage.mapping]: This seems more like `LightUpdate` of some kind

File: mappings/net/minecraft/world/World.mapping [hunk 18]
```diff
@@ -80,30 +318,192 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
...
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
+		ARG 4 blockId
+	METHOD method_245 updateWeatherCycles ()V
+	METHOD method_246 setBlockDirty (III)V
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
+	METHOD method_247 camTransferPowerInDirection (IIII)Z
```
Comment (DanyGames2014) [inline mappings/net/minecraft/world/World.mapping]: typo

File: mappings/net/minecraft/world/World.mapping [hunk 19]
```diff
@@ -80,30 +318,192 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
...
+		ARG 2 y
+		ARG 3 z
+		ARG 4 direction
+	METHOD method_248 manageChunkUpdatesAndEvents ()V
+	METHOD method_249 hasSkyLight (III)Z
 		ARG 1 x
 		ARG 2 y
 		ARG 3 z
-	METHOD method_243 (III)V
+	METHOD method_250 isEmittingRedstonePowerInDirection (IIII)Z
```
Comment (DanyGames2014) [inline mappings/net/minecraft/world/World.mapping:416]: [isEmittingRedstonePower](https://maven.fabricmc.net/docs/yarn-1.20.6+build.3/net/minecraft/world/RedstoneView.html#isEmittingRedstonePower(net.minecraft.util.math.BlockPos,net.minecraft.util.math.Direction)) in Yarn

**Review by DanyGames2014 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/block/Block.mapping [hunk 20]
```diff
@@ -180,7 +180,7 @@ CLASS net/minecraft/class_17 net/minecraft/block/Block
 		ARG 2 x
 		ARG 3 y
 		ARG 4 z
-	METHOD method_1568 isEmittingRedstonePower (Lnet/minecraft/class_14;IIII)Z
+	METHOD method_1568 isEmittingRedstonePowerInDirection (Lnet/minecraft/class_14;IIII)Z
```
Comment (DanyGames2014) [inline mappings/net/minecraft/block/Block.mapping:183]: arg5 should be direction, not meta

**Review by DanyGames2014 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/Entity.mapping [hunk 21]
```diff
@@ -28,7 +28,7 @@ CLASS net/minecraft/class_57 net/minecraft/entity/Entity
 	FIELD field_1616 dataTracker Lnet/minecraft/class_216;
 	FIELD field_1617 minBrightness F
 		COMMENT Set to 1 when rendering inside inventory.
-	FIELD field_1618 inChunk Z
+	FIELD field_1618 persistent Z
```
Comment (calmilamsy) [inline mappings/net/minecraft/entity/Entity.mapping]: Should be `isPersistent` to be consistent with other boolean method names.

**Review by calmilamsy — CHANGES_REQUESTED** [review]
Had a look, and I'm agreement with the current comments. Sort those out, and I'll deal with the merge conflicts.

File: mappings/net/minecraft/world/chunk/light/LightStorage.mapping (hunk 17 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/chunk/light/LightStorage.mapping]: sure... there's no equivalent yarn mapping, but i do agree it is more primitive so im just gonna keep it in the same place and call it lightupdate

File: mappings/net/minecraft/world/chunk/ChunkManager.mapping [hunk 22]
```diff
@@ -0,0 +1,18 @@
+CLASS net/minecraft/class_51 net/minecraft/world/chunk/ChunkManager
+	METHOD method_1801 tick ()Z
+	METHOD method_1802 hasChunk (II)Z
```
Comment (DanyGames2014) [inline mappings/net/minecraft/world/chunk/ChunkManager.mapping]: [isChunkLoaded](https://maven.fabricmc.net/docs/yarn-1.21+build.2/net/minecraft/world/chunk/ChunkManager.html#isChunkLoaded(int,int)) in Yarn

File: mappings/net/minecraft/world/chunk/ChunkManager.mapping [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+		ARG 1 x
+		ARG 2 z
+	METHOD method_1803 generate (Lnet/minecraft/class_51;II)V
```
Comment (DanyGames2014) [inline mappings/net/minecraft/world/chunk/ChunkManager.mapping]: This seems to only do decoration (Features, Trees, Snow), so `decorate` would be more fitting

File: mappings/net/minecraft/world/chunk/ChunkManager.mapping [hunk 24]
```diff
@@ -0,0 +1,18 @@
+CLASS net/minecraft/class_51 net/minecraft/world/chunk/ChunkManager
+	METHOD method_1801 tick ()Z
```
Comment (DanyGames2014) [inline mappings/net/minecraft/world/chunk/ChunkManager.mapping:2]: This seems to unload 100 chunks from the "Drop" Set, not sure if tick is accurate

**Review by DanyGames2014 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/world/chunk/ChunkManager.mapping (hunk 24 above)
Comment (DanyGames2014) [reply mappings/net/minecraft/world/chunk/ChunkManager.mapping:2]: Now that I think about it, its upto the ChunkManager to do whatever it needs soooo you can disregard this comment

**Review by DanyGames2014 — APPROVED** [review]
Chocola Approves!

File: mappings/net/minecraft/world/World.mapping [hunk 25]
```diff
@@ -20,58 +61,254 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
...
+		ARG 1 entityClass
+	METHOD method_175 collectEntitiesByClass (Ljava/lang/Class;Lnet/minecraft/class_25;)Ljava/util/List;
+		ARG 1 entityClass
+		ARG 2 box
 	METHOD method_176 getOrCreateState (Ljava/lang/Class;Ljava/lang/String;)Lnet/minecraft/class_58;
 		ARG 1 stateClass
 		ARG 2 id
+	METHOD method_177 getPlayer (Ljava/lang/String;)Lnet/minecraft/class_54;
+		ARG 1 name
 	METHOD method_178 addParticle (Ljava/lang/String;DDDDDD)V
```
Comment (Atilist) [inline mappings/net/minecraft/world/World.mapping:190]: This method does not seem to have all parameters mapped. After some testing, I can conclude that the first 3 doubles are xyz-coordinates. The other 3 would be more tricky to name due to serving 2 different purposes. They are xyz-velocity values by default except for the "reddust" particle. In that special case, they are RGB values (note that they are added on top of the existing red base colour)

Comment (Atilist) [reply mappings/net/minecraft/world/World.mapping:190]: Here are some suggestions for parameter names.
For the first 3 doubles: x y z
For the last 3 doubles: specialLogic1 specialLogic2 specialLogic3

Comment (DanyGames2014) [reply mappings/net/minecraft/world/World.mapping:190]: Could be named velocityX, velocityY and velocityZ in acoordance to Yarn as that is what its used for in most casses and add a Javadoc comment that for redstone dust its used for something different

File: mappings/net/minecraft/world/chunk/ChunkManager.mapping (hunk 24 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/chunk/ChunkManager.mapping:2]: disregarded

File: mappings/net/minecraft/world/World.mapping (hunk 25 above)
Comment (matthewperiut) [reply mappings/net/minecraft/world/World.mapping:190]: Changed to velocity with a Java doc on special behaviors

**Review by Atilist — APPROVED** [review]
The comment I made on the mappings has been addressed with a solution. I see no other things that need reviewing so I am approving this PR
